modules/3d/src/pointcloud/io_ply.cpp

```diff
@@ -83,7 +84,7 @@ bool PlyDecoder::parseHeader(std::ifstream &file)
                 auto splitArrElem = split(s, ' ');
                 if (splitArrElem[2] == "x" || splitArrElem[2] == "red" || splitArrElem[2] == "nx")
                 {
-                    if (splitArrElem[1] != "float") {
+                    if (splitArrElem[1] != "float" && splitArrElem[1] != "uchar") {
                         CV_LOG_ERROR(NULL, "Provided PLY file format '" << splitArrElem[1] << "' is not supported");
```

Comment:
Coordinates and normals are expected to be floats. Color information is expected to be uchar. Please fix the condition.